### DISCUSSION TITLE
how do i set limits

### DISCUSSION THREAD

**Discussion by NeedNot** [danthedeckie/simpleeval#105] [Q&A]
i want to use this to allow users to use their very own formula for my leveling bot so it will save people's formula as a string and then run it. ofc I want to limit it to math. 

how would I make it only work with math and make sure it doesn't cause loops or very complex resource-intensive problems that take more than a second to process

**Comment by TheTechRobo** [comment]
You can overwrite the operators with functions that limit  the max input values, just like how it's done internally with the exponent operator.

Not sure how to do it otherwise tho

**Comment by shahriyardx** [comment]
---------- comment removed by moderator ---------

**Comment by TheTechRobo** [comment]
Ok, good to know

**Comment by shahriyardx** [comment]
Not your solution. His idea is dumb. Becuse custom formula for exp is bad.

I can put a formula like old_xp + 99999 and then after getting 99999 xp change formula to old_xp + 10

So I got so many levels while others are lagging behind for no reason. So the rule should be same for all

**Comment by TheTechRobo** [comment]
I think they mean server owners can change the server's leveling speed.
